File: src/main/java/com/jss/jssbackend/modules/quotation/service/BuildingDomiciliationService.java
package com.jss.jssbackend.modules.quotation.service;

import java.util.List;

import com.jss.jssbackend.modules.quotation.model.BuildingDomiciliation;

public interface BuildingDomiciliationService {
    public List<BuildingDomiciliation> getBuildingDomiciliations();

    public BuildingDomiciliation getBuildingDomiciliation(Integer id);
}

File: client-myjss/src/app/modules/quotation/services/building.domiciliation.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../main/services/appRest.service';
import { BuildingDomiciliation } from '../model/BuildingDomiciliation';

@Injectable({
  providedIn: 'root'
})
export class BuildingDomiciliationService extends AppRestService<BuildingDomiciliation> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getBuildingDomiciliations() {
    return this.getListCached(new HttpParams(), "building-domiciliations");
  }
}
